common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java

```diff
@@ -2,14 +2,227 @@
+            classLogger.error("Failed to read data from the file.", e);
+        }
+
+        return columnNameToIndexMap;
+    }
+
+    private WindowsBatteryParsedData.WindowsBatteryMetrics getWindowsBatteryMetrics(String[] fieldValues,
+                                                                                    Map<String, Integer>columnNameToIndexMap)
+    {
+        WindowsBatteryParsedData.WindowsBatteryMetrics windowsBatteryMetrics =
+                new WindowsBatteryParsedData.WindowsBatteryMetrics();
+
+        windowsBatteryMetrics.setEnergyLoss(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[0], columnNameToIndexMap));
+        windowsBatteryMetrics.setCPUEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[1], columnNameToIndexMap));
+        windowsBatteryMetrics.setSocEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[2], columnNameToIndexMap));
+        windowsBatteryMetrics.setDisplayEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[3], columnNameToIndexMap));
+        windowsBatteryMetrics.setDiskEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[4], columnNameToIndexMap));
+        windowsBatteryMetrics.setNetworkEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[5], columnNameToIndexMap));
+        windowsBatteryMetrics.setMBBEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[6], columnNameToIndexMap));
+        windowsBatteryMetrics.setOtherEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[7], columnNameToIndexMap));
+        windowsBatteryMetrics.setEmiEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[8], columnNameToIndexMap));
+        windowsBatteryMetrics.setCPUEnergyConsumptionWorkOnBehalf(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[9], columnNameToIndexMap));
+        windowsBatteryMetrics.setCPUEnergyConsumptionAttributed(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[10], columnNameToIndexMap));
+        windowsBatteryMetrics.setTotalEnergyConsumption(
+                getOneMetric(fieldValues, WindowsBatteryParsedData.METRICS_NAME[11], columnNameToIndexMap));
+
+        windowsBatteryMetrics.setTimeStamp(getOneStringField(fieldValues, "TimeStamp", columnNameToIndexMap));
+        return windowsBatteryMetrics;
+    }
+
+    private long getOneMetric(String[] fieldValues, String metricName, Map<String, Integer>columnNameToIndexMap)
+    {
+        int index = columnNameToIndexMap.getOrDefault(metricName, -1);
+        if (index != -1)
+        {
+            return Long.parseLong(fieldValues[index]);
```

Comment:
Please make sure parseLong won't throw an exception here if the param is not in a valid float format, I suggest we should handle the exception.

Reply:
Thanks, a try-catch added.